js/src/views/Contract/Events/events.js

```diff
@@ -27,106 +27,23 @@ export default class Events extends Component {
   }
 
   static propTypes = {
-    events: PropTypes.array,
-    isTest: PropTypes.bool
-  }
-
-  state = {
-    transactions: {}
-  }
-
-  componentDidMount () {
-    this.componentWillReceiveProps(this.props);
-  }
-
-  componentWillReceiveProps (newProps) {
-    this.retrieveTransactions(newProps.events);
+    events: PropTypes.array
   }
 
   render () {
-    const { events, isTest } = this.props;
-    const { transactions } = this.state;
+    const { events } = this.props;
 
     if (!events || !events.length) {
       return null;
     }
 
-    const rows = events.map((event) => {
-      const transaction = transactions[event.transactionHash] || {};
-      const classes = `${styles.event} ${styles[event.state]}`;
-      const url = `https://${isTest ? 'testnet.' : ''}etherscan.io/tx/${event.transactionHash}`;
-      const keys = Object.keys(event.params).map((key, index) => {
-        return <div className={ styles.key } key={ `${event.key}_key_${index}` }>{ key }</div>;
-      });
-      const values = Object.values(event.params).map((value, index) => {
-        return (
-          <div className={ styles.value } key={ `${event.key}_val_${index}` }>
-            { this.renderValue(value) }
-          </div>
-        );
-      });
-
-      return (
-        <tr className={ classes } key={ event.key }>
-          <td>{ event.state === 'pending' ? 'pending' : event.blockNumber.toFormat(0) }</td>
-          <td className={ styles.txhash }>
-            <div>{ transaction.from }</div>
-            <a href={ url } target='_blank'>{ event.transactionHash }</a>
-          </td>
-          <td>
-            <div>{ event.type } =></div>
-            { keys }
-          </td>
-          <td>
-            <div>&nbsp;</div>
-            { values }
-          </td>
-        </tr>
-      );
-    });
-
     return (
       <Container>
         <ContainerTitle title='events' />
         <table className={ styles.events }>
-          <tbody>{ rows }</tbody>
+          <tbody>{ events.map((event) => <Event event={ event } key={ event.key } />) }</tbody>
```

Comment:
As these will be references, there won't be additional copies made. So at least from an overhead point of view things should be fine.

As a general point, you are probably right - no need to do connect on Events & Event if it can be helped.